core/procs.go

```diff
@@ -152,6 +152,11 @@ var procMeta Proc = func(args []Object) Object {
 		if meta != nil {
 			return meta
 		}
+	case *Type:
```

Comment:
This should not be necessary. Embedding `MetaHolder` into `Type` should make it implement `Meta`, or am I missing something?

Reply:
This one has me stumped too: if I remove the `case *Type` entirely, `(doc Int)` and such stop working.

And if I make the first one into `case Meta, *Type:`, I get this from the compiler:

```
./procs.go:151:14: obj.GetMeta undefined (type Object has no field or method GetMeta)
```

I think the combining of a non-reference with a reference type match in a single `case` means the compiler can't assume anything further about the object (`obj` in this case) than that it's of the original declared type (`Object`, based on `args []Object` and assigning it from `args[0]`).

Does any other (useful) type of object, provided as an argument to `(meta ...)`, get passed as a reference, as `Type` does (in that `ProcMeta()` gets an `obj` of `*Type`)?

Reply:
Re-reading that comment of mine, it's poorly worded, but you might figure it out anyway. Let me know if not....

Reply:
Hm... I'll look into this issue later, but the PR looks great as is. Merging. Thanks!

Reply:
Just looked into this more. `Meta` interface has two methods: `GetMeta` and `WithMeta`. `MetaHolder` only implements `GetMeta`, so embedding it is not enough to implement `Meta`. In case of `Type` it doesn't even make sense for `Type` to implement `Meta`, because calling `WithMeta` on a type is nonsense (meta data is read-only for types). So I think your implementation is right.